duktape/src/main/jni/duktape-jni.cpp

```diff
@@ -23,27 +24,31 @@
 namespace {
 
 std::unique_ptr<GlobalRef> duktapeClass;
-static jmethodID getLocalTimeZoneOffset = nullptr;
+std::function<int(double)> getTimeZoneOffset = [](double d) { return 0; };
 
 void initialize(JNIEnv* env, jclass type) {
   duktapeClass.reset(new GlobalRef(env, type));
-  getLocalTimeZoneOffset = env->GetStaticMethodID(static_cast<jclass>(duktapeClass->get()),
-                                                  "getLocalTimeZoneOffset",
-                                                  "(D)I");
+
+  auto tzClass = env->FindClass("java/util/TimeZone");
+  auto getDefaultTimeZone = env->GetStaticMethodID(tzClass, "getDefault", "()Ljava/util/TimeZone;");
+  auto getOffset = env->GetMethodID(tzClass, "getOffset", "(J)I");
+
+  const GlobalRef timeZoneClass(env, tzClass);
+  getTimeZoneOffset = [timeZoneClass, getDefaultTimeZone, getOffset](double time) {
```

Comment:
Explicitly declaring what's captured is interesting!

Reply:
Safety first and no surprises.